schemas/spacefile/spacefile.v0.schema.json

```diff
@@ -118,25 +114,42 @@
           "description": "Routes that will be available publicly",
           "type": "array",
           "minItems": 1,
+          "uniqueItems": true,
           "items": {
             "type": "string"
           }
         },
+        "public": {
+          "description": "If the Micro should be available to the public",
+          "type": "boolean",
+          "default": false
+        },
         "actions": {
           "description": "Tasks that run on triggers like a schedule",
           "type": "array",
           "minItems": 1,
+          "uniqueItems": true,
           "items": {
             "$ref": "#/definitions/action"
           }
         }
       },
-      "required": [
-        "name",
-        "src",
-        "engine"
-      ],
-      "$comment": "primary and serve required under some conditions"
+      "required": ["name", "src", "engine"],
+      "if": {
+        "properties": {
```

Comment:
You might want to add `{"required": ["engine"]}` to the if clause or else the "serve" property will be required for micro without an engine.

Reply:
Fixed in c941a57da7e806a51f18e16d6a41720944987296